update

orthagonal opened the pull request:
-handles noTags option
-allows override of username and icon
-internalErrors will report request-error errors to slack

orthagonal committed:
update

-handles noTags option
-allows override of username and icon
-internalErrors will report request-error errors to slack

jgallen23 commented on index.js:
````diff
@@ -22,33 +22,60 @@ exports.register = (server, config, next) => {
-      // it displays all nicely on slack:
-      if (!data.message) {
-        // slack uses ``` to format text like an object:
+      if (config.noTags) {
````
can you look at making this a little cleaner?  There are a lot of if config.noTags checks.  Maybe make the tags string include the `[ ]` and then if noTags, just set the whole thing to blank string and you can still include it

jgallen23 commented on index.js:
```diff
@@ -64,10 +91,16 @@ exports.register = (server, config, next) => {
     if (config.channel) {
       slackPayload.channel = config.channel;
     }
+    if (config.icon_url) {
```
can you change to iconURL?

jgallen23 commented on index.js:
```diff
@@ -94,9 +127,14 @@ exports.register = (server, config, next) => {
-  server.methods.postMessageToSlack = postMessageToSlack;
-  server.methods.postRawDataToSlack = doPost;
+  server.decorate('server', 'postMessageToSlack', postMessageToSlack);
+  server.decorate('server', 'postToSlack', postToSlack);
```
can we change to `slackPostMessage` and `slackPostRawMessage`?

jgallen23 commented on test/server-test.js:
```diff
@@ -55,15 +58,55 @@ lab.test('lets you post an object with a special "message" field', (done) => {
+lab.test('will user a supplied username', (done) => {
+  server.stop( () => {
+    delete options.noTags;
+    options.username = "Jared";
```
;) clever

orthagonal committed:
cleaned up notags

cleaned up the config.noTags logic
changed some of the variable names for clarity
